File: e2e-pytest-playwright/pages/passenger_page.py
from pages.abstract_page import AbstractPage

class PassengerPage(AbstractPage):

  def enter_first_name(self, passenger_first_name):
    self.driver.locator("[name='passengerFirstName']").fill(passenger_first_name)

  def enter_last_name(self, passenger_last_name):
    self.driver.locator("[name='passengerLastName']").fill(passenger_last_name)

  def click_next(self):
    self.driver.locator("//input[@value='Next']").click() 

File: e2e-playwright-test-refactored/pages/passenger_page.js
const AbstractPage = require('./abstract_page');

class PassengerPage extends AbstractPage {

  constructor(page) {
    super(page);
  }
  
  async enterFirstName(firstName) {
    await this.page.locator("[name=passengerFirstName]").fill(firstName)
  }
      
  async enterLastName(lastName) {
    await this.page.locator("[name=passengerLastName]").fill(lastName)
  }
    
  async clickNext() {
    await this.page.click("input:has-text('Next')")
    this.sleep(0.5)
  }
 
};

module.exports = PassengerPage;
